src/lib389/lib389/cli_conf/replication.py

```diff
@@ -183,6 +183,13 @@ def enable_replication(inst, basedn, log, args):
         # rid is good add it to the props
         repl_properties['nsDS5ReplicaId'] = args.replica_id
 
+    # Validate consumer and hub settings
+    elif role == "consumer" or role == "hub":
+        # Check Replica ID
+        if args.replica_id is not None or args.replica_id != 65535:
```

Comment:
This introduced a regression: I can no longer create consumer or hub, it fails with:
```
/usr/sbin/dsconf -j ldapi://%2fvar%2frun%2fslapd-M2.socket replication enable --suffix=dc=example,dc=com --role=consumer --bind-dn="cn=replication manager,cn=config"
{
    "desc": "Replica ID cannot be specified for consumer and hub roles"
}
```

I think `or` should be replaced with `and`, because if `args.replica_id` is `None`, the second part of the condition will evaluate to `True`:
```
(Pdb) print(args.replica_id)
None
(Pdb) print(args.replica_id != 65535)
True
(Pdb) print(args.replica_id is not None or args.replica_id != 65535)
True
```

Reply:
I agree, the 'or' should be replaced with a 'and'. The only allowed value is 65535.